packages/contracts/src/EmailAuthSigner.sol

```diff
@@ -0,0 +1,188 @@
+    address public verifierAddr;
+    /// The templateId of the sign hash command.
+    uint256 public templateId;
+
+    constructor() {}
+
+    /// @notice Initialize the contract with an initial owner, account salt, DKIM registry address, and verifier address.
+    /// @param _initialOwner The address of the initial owner.
+    /// @param _accountSalt The account salt to derive CREATE2 address of this contract.
+    /// @param _dkimRegistryAddr The address of the DKIM registry contract.
+    /// @param _verifierAddr The address of the verifier contract.
+    /// @param _templateId The templateId of the sign hash command.
+    function initialize(
+        address _initialOwner,
+        bytes32 _accountSalt,
+        address _dkimRegistryAddr,
+        address _verifierAddr,
+        uint256 _templateId
+    ) public initializer {
+        __Ownable_init(_initialOwner);
+        accountSalt = _accountSalt;
+        templateId = _templateId;
+        require(
+            _dkimRegistryAddr != address(0),
+            "invalid dkim registry address"
+        );
+        require(
+            address(dkimRegistryAddr) == address(0),
+            "dkim registry already initialized"
+        );
+        dkimRegistryAddr = _dkimRegistryAddr;
+        emit DKIMRegistryUpdated(_dkimRegistryAddr);
+
+        require(_verifierAddr != address(0), "invalid verifier address");
+        require(
+            address(verifierAddr) == address(0),
+            "verifier already initialized"
+        );
+        verifierAddr = _verifierAddr;
+        emit VerifierUpdated(_verifierAddr);
+    }
+
+    /// @notice Updates the address of the DKIM registry contract.
+    /// @param _dkimRegistryAddr The new address of the DKIM registry contract.
+    function updateDKIMRegistry(address _dkimRegistryAddr) public onlyOwner {
+        require(
+            _dkimRegistryAddr != address(0),
+            "invalid dkim registry address"
+        );
+        dkimRegistryAddr = _dkimRegistryAddr;
+        emit DKIMRegistryUpdated(_dkimRegistryAddr);
+    }
+
+    /// @notice Updates the address of the verifier contract.
+    /// @param _verifierAddr The new address of the verifier contract.
+    function updateVerifier(address _verifierAddr) public onlyOwner {
+        require(_verifierAddr != address(0), "invalid verifier address");
+        verifierAddr = _verifierAddr;
+        emit VerifierUpdated(_verifierAddr);
+    }
+
+    /// @notice Authenticate the email sender and authorize the message in the email command.
+    /// @dev This function only verifies the authenticity of the email and command, without
+    /// handling replay protection. The calling contract should implement its own mechanisms
+    /// to prevent replay attacks, similar to how nonces are used with ECDSA signatures.
+    /// @param emailAuthMsg The email auth message containing all necessary information for authentication.
+    function authEmail(EmailAuthMsg memory emailAuthMsg) public {
```

Comment:
Related to above comment on 1271 validation - thinking that verifySignature is executing most of this logic